src/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -149,6 +149,88 @@ public static void CreateUncompressedArchive()
             }
         }
 
+        [Theory]
+        [InlineData("minimalTwoFiles")]
+        public static void CreateNormal_VerifyDataDescriptor(string folder)
+        {
+            void CreateEntry(ZipArchive archive, string fileName, string fileContents)
+            {
+                ZipArchiveEntry entry = archive.CreateEntry(fileName);
+                using (StreamWriter writer = new StreamWriter(entry.Open()))
+                {
+                    writer.Write(fileContents);
+                }
+            }
+
+            string fileName = folder + ".zip";
+
+            if (File.Exists(fileName))
+            {
+                File.Delete(fileName);
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Create))
+            {
+                // We need an unseekable stream so the data descriptor bit is turned on when saving
+                var wrappedStream = new WrappedStream(fileStream, true, true, false, null);
+
+                // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+                using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+                {
+                    CreateEntry(archive, "A", "xxx");
+                    CreateEntry(archive, "B", "yyy");
+                }
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Open))
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    fileStream.CopyTo(memoryStream);
+                    byte[] fileBytes = memoryStream.ToArray();
+
+                    // We expect the data descriptor to be on in the first local file header after creation
+                    Assert.Equal(8, fileBytes[6]);
```

Comment:
I'm sure your right (I know nothing about zips and this code) but I tried to check using https://users.cs.jmu.edu/buchhofp/forensics/formats/pkzip.html and that suggested to me that the flags are in bytes 8 and 9 not 6 and 7.